libs/net/dal/Services/ContentService.cs

```diff
@@ -489,5 +489,19 @@ public IEnumerable<NotificationInstance> GetNotificationsFor(long contentId)
         return this.Context.NotificationInstances
             .Where(n => n.ContentId == contentId);
     }
+
+    /// <summary>
+    /// Update the specified 'entity' in the database.
+    /// </summary>
+    /// <param name="entity"></param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    /// TODO: Switch to not found exception throughout services.
+    public Content UpdateStatusOnly(long contentId, ContentStatus newStatus)
```

Comment:
The main issue here is you're not accounting for concurrency issues.  Is this intentional?  If so why?  In most cases we enforce optimistic concurrency.